test/Table_Tests/src/Database/Postgres_Spec.enso

```diff
@@ -121,19 +121,47 @@ run_tests connection pending=Nothing =
 table_spec =
     db_name = Environment.get "ENSO_DATABASE_TEST_DB_NAME"
     db_host_port = (Environment.get "ENSO_DATABASE_TEST_HOST").if_nothing "localhost" . split ':'
+    db_host = db_host_port.at 0
+    db_port = if db_host_port.length == 1 then 5432 else Integer.parse (db_host_port.at 1)
     db_user = Environment.get "ENSO_DATABASE_TEST_DB_USER"
     db_password = Environment.get "ENSO_DATABASE_TEST_DB_PASSWORD"
+    ca_cert_file = Environment.get "ENSO_DATABASE_TEST_CA_CERT_FILE"
+
+    ssl_pending = if ca_cert_file.is_nothing then "PostgreSQL SSL test not configured." else Nothing
+    Test.group "[PostgreSQL] Table.aggregate should correctly infer result types" pending=ssl_pending <|
+        Test.specify "should connect without ssl parameter" <|
+            Database.connect (Postgres db_host db_port db_name (Credentials db_user db_password)) . should_succeed
+
+        Test.specify "should connect be able to require SSL" <|
+            Database.connect (Postgres db_host db_port db_name (Credentials db_user db_password) use_ssl=Require) . should_succeed
+
+        Test.specify "should connect be able to verify the certificate" <|
+            Database.connect (Postgres db_host db_port db_name (Credentials db_user db_password) use_ssl=(Verify_CA ca_cert_file)) . should_succeed
+
+            ## Default certificate should not accept the self signed certificate.
+            ca_fail = Database.connect (Postgres db_host db_port db_name (Credentials db_user db_password) use_ssl=Verify_CA)
+            ca_fail.is_error . should_equal True
+            ca_fail.is_a Sql_Error . should_equal True
+
+        Test.specify "should connect be able to verify the host name against the certificate" <|
+            Database.connect (Postgres db_host db_port db_name (Credentials db_user db_password) use_ssl=(Full_Verification ca_cert_file)) . should_succeed
+
+            ## Try connecting to localhost and then should fail verification.
+            if db_host == "127.0.0.1" then
+                ca_fail = Database.connect (Postgres "localhost" db_port db_name (Credentials db_user db_password) use_ssl=(Full_Verification ca_cert_file))
+                ca_fail.is_error . should_equal True
+                ca_fail.is_a Sql_Error . should_equal True
```

Comment:
Shouldn't this be

```suggestion
                ca_fail.catch.is_a Sql_Error . should_equal True
```

(I don't think a dataflow error reacts to `is_a` as you'd expect because it is instead an instance of the `DataflowError` builtin, it is the payload that we want to check for the type and `.catch` gives us access to the payload)